### PR TITLE
Implement new query for inventory status

### PR DESCRIPTION
**Pull request by lesser1**
- Replace query in inventory.graphql
- Extend getProductForBuyer query in catalog.graphql to also include the totalCount of product items in stock
- Refactor ProductView.view: Use new query, rename some refs, add helper function

[Gropius](https://frontend.gropius.duckdns.org/components/9d12d6ac-e85a-407c-bdff-a8321fb75d3a/issues/c4dabc69-bc35-40fc-9990-6750b5bf4c60)

### Definition of Done
- [x] Requirements of the issue are met
- [x] Code has been reviewed
- [x] Code is documented

### CODE REVIEW

File: src/views/ProductView.vue [hunk 1]
```diff
@@ -744,173 +750,271 @@ function goToWishlists() {
...
-        const nodes =
-            getInventoryStatusOfProductItemsQuery.value
-                .productItemsByProductVariant.nodes
-        if (nodes !== null && nodes !== undefined) {
-            return nodes
+const productItemsCountOfReservedItems = asyncComputed(
+    async () => {
+        trigger.value
+
+        const queryResponse = await getProductItemsCountOfInventoryStatus(
```
Comment (legendofa) [inline src/views/ProductView.vue]: How about putting the part which is duplicated in multiple locations in a function?

File: src/views/ProductView.vue [hunk 2]
```diff
@@ -744,173 +750,271 @@ function goToWishlists() {
...
-const numberOfReturnedProductItems = computed(() => {
-    if (inventoryStatusOfProductItems.value !== null) {
-        return inventoryStatusOfProductItems.value.filter(
-            (productItem) =>
-                productItem.inventoryStatus === ProductItemStatus.Returned
-        ).length
+async function getProductItemsCountOfInventoryStatus(
+    inventoryStatus: ProductItemStatus
+): Promise<GetProductItemsCountOfInventoryStatusQuery | null> {
+    if (inventoryStatus == undefined) {
```
Comment (legendofa) [inline src/views/ProductView.vue:959]: Not sure about Typescript, but it this the best way to solve this? How about an Optional?

**Review by legendofa — CHANGES_REQUESTED** [review]

File: src/views/ProductView.vue (hunk 2 above)
Comment (lesser1) [reply src/views/ProductView.vue:959]: This is the way you define optionals as return types in TS. You can use the '?' syntax for defining members of interfaces or classes but not for the return types.

File: src/views/ProductView.vue (hunk 1 above)
Comment (lesser1) [reply src/views/ProductView.vue]: 67ad16477c1e8d4e7168d03ba2e51415e30ba075

**Review by legendofa — APPROVED** [review]

**Pull request merged**